pkg/instancetype/instancetype.go

```diff
@@ -522,17 +523,56 @@ func (m *methods) findClusterInstancetype(vm *virtv1.VirtualMachine) (*instancet
 	return instancetype, nil
 }
 
+func resolveConflicts(field *k8sfield.Path, requestType k8sv1.ResourceName, vmiSpec *virtv1.VirtualMachineInstanceSpec, instancetypeSpec *instancetypev1alpha2.VirtualMachineInstancetypeSpec) (Conflicts, bool) {
+	hasGuestField := false
+
+	switch requestType {
+	case k8sv1.ResourceCPU:
+		if vmiSpec.Domain.CPU != nil {
+			return Conflicts{field.Child("domain", "cpu")}, true
+		}
+		if instancetypeSpec.CPU.Guest != 0 {
+			hasGuestField = true
+		}
+	case k8sv1.ResourceMemory:
+		if vmiSpec.Domain.Memory != nil {
+			return Conflicts{field.Child("domain", "memory")}, true
+		}
+		if !instancetypeSpec.Memory.Guest.IsZero() {
+			hasGuestField = true
+		}
+	}
+
+	if _, hasRequests := instancetypeSpec.Resources.Requests[requestType]; hasRequests && hasGuestField {
+		return Conflicts{field.Child("domain", "resources", "requests", string(requestType))}, true
+	}
+
+	if _, hasRequests := vmiSpec.Domain.Resources.Requests[requestType]; hasRequests {
+		return Conflicts{field.Child("domain", "resources", "requests", string(requestType))}, true
+	}
+
+	if _, hasLimits := vmiSpec.Domain.Resources.Limits[requestType]; hasLimits {
+		return Conflicts{field.Child("domain", "resources", "limits", string(requestType))}, true
+	}
+
+	return nil, false
+}
+
 func applyCpu(field *k8sfield.Path, instancetypeSpec *instancetypev1alpha2.VirtualMachineInstancetypeSpec, preferenceSpec *instancetypev1alpha2.VirtualMachinePreferenceSpec, vmiSpec *virtv1.VirtualMachineInstanceSpec) Conflicts {
 	if vmiSpec.Domain.CPU != nil {
 		return Conflicts{field.Child("domain", "cpu")}
 	}
 
-	if _, hasCPURequests := vmiSpec.Domain.Resources.Requests[k8sv1.ResourceCPU]; hasCPURequests {
-		return Conflicts{field.Child("domain", "resources", "requests", string(k8sv1.ResourceCPU))}
+	if conflicts, hasConflicts := resolveConflicts(field, k8sv1.ResourceCPU, vmiSpec, instancetypeSpec); hasConflicts {
+		return conflicts
 	}
 
-	if _, hasCPULimits := vmiSpec.Domain.Resources.Limits[k8sv1.ResourceCPU]; hasCPULimits {
-		return Conflicts{field.Child("domain", "resources", "limits", string(k8sv1.ResourceCPU))}
+	if instancetypeSpec.CPU.Guest == 0 {
```

Comment:
Again I'd rather enforce the expected schema (guest CPU or resource CPU but not both defined in the instance type) here through a validation webhook and just have the conditional here focus on `instancetypeSpec.Resources.Requests[k8sv1.ResourceCPU]` being present.